fml/message_loop.cc

```diff
@@ -0,0 +1,59 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/message_loop.h"
+#include <utility>
+#include "flutter/fml/message_loop_impl.h"
+#include "flutter/fml/task_runner.h"
+#include "flutter/fml/thread_local.h"
+#include "lib/ftl/memory/ref_counted.h"
+#include "lib/ftl/memory/ref_ptr.h"
+
+namespace fml {
+
+FML_THREAD_LOCAL ThreadLocal TLSMessageLoop([](intptr_t value) {
+  delete reinterpret_cast<MessageLoop*>(value);
+});
+
+MessageLoop& MessageLoop::GetCurrent() {
+  auto loop = reinterpret_cast<MessageLoop*>(TLSMessageLoop.Get());
+  if (loop != nullptr) {
+    return *loop;
+  }
+
+  loop = new MessageLoop();
```

Comment:
I'm not sure we want an auto-initializing loop.  The way we've done this before is to have the MessageLoop constructor set the TLS entry (and CHECK that the TLS entry hasn't been set before).  That means you have to manually set up the loop, but it makes the startup process more orderly.